Makefile

```diff
@@ -85,13 +87,21 @@ go-version-check:
 		[ $(GO_VERSION_MAJ) -eq 1 -a $(GO_VERSION_MIN) -ge 12 ] || (echo "FATAL: Go version should be >= 1.12.x" ; exit 1 ; )
 
 .PHONY: lint
-lint:
+lint: deps
 	# explicitly enable GO111MODULE otherwise go mod will fail
 	GO111MODULE=on go mod tidy && GO111MODULE=on go mod vendor && GO111MODULE=on go mod verify
 	$(GO) vet ./...
 	test -z `$(GOFMT) -l $(SKUBA_SRCS)` || { $(GOFMT) -d $(SKUBA_SRCS) && false; }
 	$(TERRAFORM) fmt -check=true -write=false -diff=true ci/infra
-	find ci -type f -name "*.sh" | xargs bashate
+	$(BINPATH)/golangci-lint run cmd/... pkg/... internal/...
+	find ci -type f -name "*.sh" | xargs $(BINPATH)/shellcheck
+
+.PHONY: deps
+deps:
+	curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s -- -b $(BINPATH) v1.21.0
```

Comment:
Do we want to pin the `master` to a dedicated version/commit too?

Reply:
this is the recommended procedure in the upstream golangci-lint repo, the binary is pinned but I think that they will be monitoring the install script.
Let's keep it this way and see what happens